service/sharddistributor/store/etcd/executorstore/shardcache/namespaceshardcache.go

```diff
@@ -5,58 +5,60 @@ import (
 	"fmt"
 	"strings"
 	"sync"
+	"time"
 
 	clientv3 "go.etcd.io/etcd/client/v3"
 
+	"github.com/uber/cadence/common/clock"
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
 	"github.com/uber/cadence/service/sharddistributor/store"
+	"github.com/uber/cadence/service/sharddistributor/store/etcd/etcdclient"
 	"github.com/uber/cadence/service/sharddistributor/store/etcd/etcdkeys"
 	"github.com/uber/cadence/service/sharddistributor/store/etcd/etcdtypes"
 	"github.com/uber/cadence/service/sharddistributor/store/etcd/executorstore/common"
 )
 
+// namespaceRefreshLoopWatchInterval is the interval between watch retries in the namespace refresh loop
+const namespaceRefreshLoopWatchRetryInterval = 150 * time.Millisecond
+
 type namespaceShardToExecutor struct {
 	sync.RWMutex
 
-	shardToExecutor     map[string]*store.ShardOwner   // shardID -> shardOwner
-	shardOwners         map[string]*store.ShardOwner   // executorID -> shardOwner
-	executorState       map[*store.ShardOwner][]string // executor -> shardIDs
-	executorRevision    map[string]int64
-	namespace           string
-	etcdPrefix          string
-	changeUpdateChannel clientv3.WatchChan
-	stopCh              chan struct{}
-	logger              log.Logger
-	client              *clientv3.Client
-	pubSub              *executorStatePubSub
+	shardToExecutor  map[string]*store.ShardOwner   // shardID -> shardOwner
+	shardOwners      map[string]*store.ShardOwner   // executorID -> shardOwner
+	executorState    map[*store.ShardOwner][]string // executor -> shardIDs
+	executorRevision map[string]int64
+	namespace        string
+	etcdPrefix       string
+	stopCh           chan struct{}
+	logger           log.Logger
+	client           etcdclient.Client
```

Comment:
I think it looks strange only to introduce an interface here. I think we should either introduce it everywhere (so only the interface depends the clientv3 package), or we should not do it.